.github/workflows/package-helm.yaml

```diff
@@ -25,14 +25,43 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+     -  name: Import GPG key
+        uses: crazy-max/ghaction-import-gpg@v6
+        with:
+          gpg_private_key: ${{ secrets.BOT_GPG_PRIVATE_KEY }}
+          passphrase: ${{ secrets.BOT_PASSPHRASE }}
+
+      - name: Export GPG private key
+        run: |
+          gpg --batch \
+              --yes \
+              --pinentry-mode loopback \
+              --passphrase ${{ secrets.BOT_PASSPHRASE }} \
+              --export-secret-keys ${{ secrets.BOT_GPG_ID }} \
+              > ~/.gnupg/secring.gpg
+
+      - name: Verify GPG secret key file
+        run: |
+          if [ ! -f ~/.gnupg/secring.gpg ]; then
+            echo "Error: GPG secret key file '~/.gnupg/secring.gpg' not found!" >&2
+            exit 1
+          fi
+
       - name: Install Helm
         uses: azure/setup-helm@v4.0.0
 
       - name: Package Stable Helm Charts
         run: |
           find "$CHARTS_DIR" -name values.yaml | xargs -I '{}' \
               sed -e s"/pullPolicy:.*/pullPolicy: IfNotPresent/" -i '{}'
-          helm package --version "$GITHUB_REF_NAME" --app-version "$GITHUB_REF_NAME" "$CHARTS_DIR"/*
+          echo ${{ secrets.BOT_PASSPHRASE }} | helm package \
+            --sign \
+            --key ${{ secrets.BOT_GPG_ID }} \
```

Comment:
In my local test, this does not work. Looks like helm requires the uid of the key, so AFAICT in this case that would be `nripluginsbot@yahoo.com`.

Reply:
So locally, with my normal keyring in use which has both the bot's main key and the subkeys present, the only way how I got helm to successfully sing the chart package with the signing subkey was to:
- editing the main key, removing the signing capability (leaving only key certification)
- then using `nripluginsbot@yahoo.com` uid for `--key`

Otherwise if there are multiple keys, since there is no way to force stupid Helm to use the correct one, it always tried to sign using the master key.

Reply:
> So locally, with my normal keyring in use which has both the bot's main key and the subkeys present, the only way how I got helm to successfully sing the chart package with the signing subkey was to:
> 
> - editing the main key, removing the signing capability (leaving only key certification)
> 
> - then using `nripluginsbot@yahoo.com` uid for `--key`
> 
> 
> 
> Otherwise if there are multiple keys, since there is no way to force stupid Helm to use the correct one, it always tried to sign using the master key.

We can export a specific secret key and use that particular keyring with Helm?

Reply:
Hey @klihub .
I've tested using the subkey with Helm, and it worked successfully. Regarding the value for the` --key `flag, you’re correct - I tested it with the email address, and it passed.

You can see the example run [here](https://github.com/fmuyassarov/nri-plugins/actions/runs/12928042336)  and the workflow I used [here](https://github.com/fmuyassarov/nri-plugins/actions/runs/12928042336/workflow). 

Based on the result above, I have now updated the PR. 
When dealing with multiple keys, it can be challenging indeed if several GPG keys share the same email address. However, for the purpose workflow, which is based on a single GPG key (subkey), this won't be an issue (I hope).

Reply:
> Based on the result above, I have now updated the PR. When dealing with multiple keys, it can be challenging indeed if several GPG keys share the same email address. However, for the purpose of the workflow, which is based on a single GPG key (subkey), this won't be an issue (I hope).

It's a bit worth than that because if you have a signing-only subkey but you haven't removed the signing capability from your main key, there is no way to select which one Helm will use (picked by uid vs. instead of fingerprint), and AFAICT it then will then try to use the main one. But this should not be a problem for us, because I removed signing from the bots main key.